### PR TITLE
Update users

### PR DESCRIPTION
**Pull request by lannp**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -32,6 +35,13 @@ def authenticated? remember_token
   def forget
     update_attributes! remember_digest: nil
   end
+
+  private
+  def avatar_size
+    if avatar.size > Settings.avatar.image_storage.megabytes
+      errors.add(:avatar, t("edit_user.p_warning"))
```
Comment (ThuBM) [inline app/models/user.rb]: chuyển thành error.add ..... if

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<li>
+  <%= image_tag user.avatar.url,
+      size: Settings.avatar.image_size if user.avatar %>
```
Comment (ThuBM) [inline app/views/users/_user.html.erb]: indent

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 class User < ActiveRecord::Base
   attr_accessor :remember_token
   before_save {self.email = email.downcase}
+  mount_uploader :avatar, AvatarUploader
+  validate  :avatar_size
```
Comment (ThuBM) [inline app/models/user.rb]: remove 1 space before :

File: app/models/user.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
+
   validates :name,  presence: true, length: {maximum: Settings.maximum_name_length}
```
Comment (ThuBM) [inline app/models/user.rb]: remove 1 space before presence

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:edit, :update]
+  before_action :correct_user,   only: [:edit, :update]
```
Comment (ThuBM) [inline app/controllers/users_controller.rb]: remove 2 spaces before only:

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -1,12 +1,12 @@
-<% if @user.errors.any? %>
+<% if object.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      <%= t "error_message" %> <%= pluralize @user.errors.count, t("error") %>.
+      The form contains <%= pluralize(object.errors.count, "error") %>.
```
Comment (ThuBM) [inline app/views/shared/_error_messages.html.erb]: remove ()

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      <%= f.password_field :password, class: "form-control" %>
+      <%= f.label :password_confirmation, t("edit_user.confir") %>
+      <%= f.password_field :password_confirmation, class: "form-control" %>
+        <div class="row">
+          <div class="col-md-5">
+            <%= image_tag @user.avatar.url, size: Settings.avatar.image_size_edit %>
+          </div>
+          <div class="col-md-7">
+            <span class="avatar">
+              <%= f.file_field :avatar, accept: Settings.image_type_accept,onchange: "PreviewImage('user_image');" %>
```
Comment (trantube94) [inline app/views/users/edit.html.erb]: add space

Comment (ThuBM) [inline app/views/users/edit.html.erb]: Split to 2 lines

**Comment by ThuBM** [comment]
OK

File: app/views/users/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,29 @@
+<% provide:title, t("edit_user.edit") %>
+<h1><%= t "edit_user.update" %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (chuanhtuan) [inline app/views/users/edit.html.erb]: ()

**Pull request merged**